src/main/java/gov/cms/mat/cql_elm_translation/controllers/EffectiveDataRequirementController.java

```diff
@@ -0,0 +1,83 @@
+package gov.cms.mat.cql_elm_translation.controllers;
+
+import gov.cms.mat.cql_elm_translation.exceptions.ResourceNotFoundException;
+import gov.cms.mat.cql_elm_translation.service.EffectiveDataRequirementService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.collections4.CollectionUtils;
+
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Library;
+import org.hl7.fhir.r4.model.Resource;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.security.Principal;
+import java.util.Optional;
+
+@Slf4j
+@RestController
+@RequiredArgsConstructor
+public class EffectiveDataRequirementController {
+  private final EffectiveDataRequirementService effectiveDataRequirementService;
+
+  @PutMapping(
+      value = "/effective-data-requirements",
+      produces = {MediaType.APPLICATION_JSON_VALUE},
+      consumes = {MediaType.TEXT_PLAIN_VALUE})
+  public ResponseEntity<String> getEffectiveDataRequirements(
+      Principal principal,
+      @RequestBody String bundleStr,
+      @RequestHeader("Authorization") String accessToken,
+      @RequestParam(required = true, name = "libraryName") String libraryName,
+      @RequestParam(required = true, name = "measureId") String measureId) {
```

Comment:
This is fine for now but maybe we can improve this in the future. we can pass just Measure resource or expression list and CQL string only. we do not need the whole bundle.

Reply:
@adongare get effective data requirements seems to need both measure and library?

Reply:
it needs just expressions(populations, SDEs, stratifications) which is part of measure resource and CQL. So, either we can pass measure resource and CQL or just list of these expressions and measure cql. measure cql will be translated by the translator and fetch included libraries. then build the effectiveDr based on expressions. but let's not handle it now. will create a ticket for it.

Reply:
Ok I see, thanks so much for the info! @adongare